File: BOJ/2748.js
const scan = require('fs').readFileSync('/dev/stdin').toString();
let cache = new BigInt64Array(91);
cache[0] = 0n, cache[1] = 1n, cache[2] = 1n;
const fib = (n) => {
    if (cache[n]) return cache[n];
    else {
        return cache[n] = BigInt(fib(n-1) + fib(n-2));
    }
}
console.log(fib(parseInt(scan, 10)).toString().replace('n', ''));

File: BOJ/2748.cpp
#include <iostream>
long long int fib(long long int cache[], int n) {
    if (cache[n]) return cache[n];
    else {
        cache[n] = fib(cache, n-1) + fib(cache, n-2);
        return cache[n];
    }
}

int main() {
    long long int cache[91] = {0, 1, 1}; int n;
    std::cin >> n;
    printf("%lld", fib(cache, n));
    return 0;
}